deepsensor/model/model.py

```diff
@@ -621,6 +621,342 @@ def unnormalise_pred_array(arr, **kwargs):
         return pred
 
 
+    def predict_patch(
+        self,
+        tasks: Union[List[Task], Task],
+        X_t: Union[
+            xr.Dataset,
+            xr.DataArray,
+            pd.DataFrame,
+            pd.Series,
+            pd.Index,
+            np.ndarray,
+        ],
+        data_processor: Union[
+            xr.DataArray,
+            xr.Dataset,
+            pd.DataFrame,
+            List[Union[xr.DataArray, xr.Dataset, pd.DataFrame]],
+        ],
+        stride_size: Union[float, tuple[float]],
+        patch_size: Union[float, tuple[float]],
+        X_t_mask: Optional[Union[xr.Dataset, xr.DataArray]] = None,
+        X_t_is_normalised: bool = False,
+        aux_at_targets_override: Union[xr.Dataset, xr.DataArray] = None,
+        aux_at_targets_override_is_normalised: bool = False,
+        resolution_factor: int = 1,
+        pred_params: tuple[str] = ("mean", "std"),
+        n_samples: int = 0,
+        ar_sample: bool = False,
+        ar_subsample_factor: int = 1,
+        unnormalise: bool = False,
+        seed: int = 0,
+        append_indexes: dict = None,
+        progress_bar: int = 0,
+        verbose: bool = False,
+
+    ) -> Prediction:
+        """
+        Predict on a regular grid or at off-grid locations.
+
+        Args:
```

Comment:
Missing argument descriptions in the docstring for `predict_patch`: `ar_subsample_factor`, `data_processor`